src/main/kotlin/LocalEvaluationClient.kt

```diff
@@ -67,15 +94,108 @@ class LocalEvaluationClient internal constructor(
 
     @JvmOverloads
     fun evaluateV2(user: ExperimentUser, flagKeys: Set<String> = setOf()): Map<String, Variant> {
-        val flagConfigs = flagConfigService.getFlagConfigs().toMap()
+        val flagConfigs = flagConfigStorage.getFlagConfigs()
         val sortedFlagConfigs = topologicalSort(flagConfigs, flagKeys)
         if (sortedFlagConfigs.isEmpty()) {
             return mapOf()
         }
-        val evaluationResults = evaluation.evaluate(user.toEvaluationContext(), sortedFlagConfigs)
-        Logger.d("evaluate - user=$user, result=$evaluationResults")
+        val enrichedUser = enrichUser(user, sortedFlagConfigs)
+        val evaluationResults = wrapMetrics(
+            metric = metrics::onEvaluation,
+            failure = metrics::onEvaluationFailure,
+        ) {
+            evaluation.evaluate(enrichedUser.toEvaluationContext(), sortedFlagConfigs)
+        }
+        Logger.d("evaluate - user=$enrichedUser, result=$evaluationResults")
         val variants = evaluationResults.toVariants()
         assignmentService?.track(Assignment(user, variants))
         return variants
     }
+
+    private fun enrichUser(user: ExperimentUser, flagConfigs: List<EvaluationFlag>): ExperimentUser {
+        val groupedCohortIds = flagConfigs.getGroupedCohortIds()
+        if (cohortStorage == null) {
+            if (groupedCohortIds.isNotEmpty()) {
+                Logger.e("Local evaluation flags target cohorts but cohort targeting is not configured.")
```

Comment:
nit: can be more explicit with logging during `evaluate`, e.g. log which flags do not have the required cohorts in storage